lib/api/core/models/repositories/repository.js

```diff
@@ -269,19 +264,14 @@ class Repository {
    *   key: if provided, stores the object to the given key instead of the default one (<collection>/<id>)
    *   ttl: if provided, overrides the default ttl set on the repository for the current operation.
    *
-   * @param {object} object the object to persist
-   * @param {object} [opts] optional options for the current operation
+   * @param {object} object - The object to persist
+   * @param {object} [options] - Optional options for the current operation
    * @returns {Promise}
    */
-  persistToCache (object, opts) {
-    const
-      options = opts || {},
-      key = options.key || this.cacheKeyPrefix + object._id;
-    let ttl = this.ttl;
-
-    if (options.ttl !== undefined) {
-      ttl = options.ttl;
-    }
+  persistToCache (object, options = {}) {
+    const ttl = options.ttl !== undefined ? options.ttl : this.ttl;
+    let key = options.key || this.getCacheKey(object._id, this.collection);
```

Comment:
`collection` parameter is not used in `getCacheKey` method
(same remark for all other calls of this within the file)